packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts

```diff
@@ -0,0 +1,195 @@
+import { DELIMITER_AFTER, DELIMITER_BEFORE, DelimiterType } from './constants';
+import { isDelimiter } from './isDelimiter';
+import {
+    ChangeSource,
+    EntityOperation,
+    PluginEventType,
+    PositionType,
+} from 'roosterjs-editor-types';
+import {
+    commitEntity,
+    createRange,
+    getEntityFromElement,
+    getEntitySelector,
+    isBlockElement,
+    matchesSelector,
+    Position,
+    safeInstanceOf,
+} from 'roosterjs-editor-dom';
+import type { Entity, EntityOperationEvent, IEditor, PluginEvent } from 'roosterjs-editor-types';
+
+const ZERO_WIDTH_SPACE = '\u200B';
+
+let delimiterId = 0;
+
+export function inlineEntityOnPluginEvent(event: PluginEvent, editor: IEditor) {
+    switch (event.eventType) {
+        case PluginEventType.EntityOperation:
+            const { wrapper } = event.entity;
+            switch (event.operation) {
+                case EntityOperation.NewEntity:
+                    // If the new entity is Read Only and inline, add delimiters to it
+                    if (isReadOnly(event.entity)) {
+                        const element = wrapper as HTMLElement;
```

Comment:
I feel this should be part of the insertEntity() API, so that it directly inserts a complete entity (an entity with two space SPANs). NewEntity event is used for other plugins to know an entity is added.

Reply:
I think we should just stop making the Empty spans Entities because of your comment above. So, I think we can still add the empty spans EntityOperation.NewEntity, 
That way we keep all the logic in Core Plugin, is that okay?

Reply:
Since all plugins are equivalent, when we add the SPANs inside a plugin, that means it is possible some other plugin runs earlier than here so they will get an entity with out these SPANs. In most cases that should be fine, but I still feel we should make the entity "fully ready" when other plugins got it, in that case plugin is not a good place to handle it.

Since all entities need to be inserted using insertEntity(), so that function should be the best place to insert the SPAN.

And there is a scenario that we will have entity but insertEntity() is not called: load existing content that contains entities. There are 2 options for that case:
1. Just do not handle it and let it be.
2. Add the SPANs here. If we go with 2, we can move the code of inserting SPANs to roosterjs-editor-dom package and wrap as a public function, then call it from both here and insertEntity()

For now, I think 1 should be fine.